notes/stv/20221014-test-benchmarker-01.txt

```diff
@@ -0,0 +1,325 @@
+#
+# <meta:header>
+#   <meta:licence>
+#     Copyright (c) 2022, ROE (http://www.roe.ac.uk/)
+#
+#     This information is free software: you can redistribute it and/or modify
+#     it under the terms of the GNU General Public License as published by
+#     the Free Software Foundation, either version 3 of the License, or
+#     (at your option) any later version.
+#
+#     This information is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU General Public License for more details.
+#
+#     You should have received a copy of the GNU General Public License
+#     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#   </meta:licence>
+# </meta:header>
+#
+
+
+
+    Target:
+
+        Test Spectra notebook & public examples before doing a PR (2022/10/13)
+
+    Result:
+
+        Success (*) 
+         (*) Benchmark runs but HDBSCAN fails to import
+
+
+# -----------------------------------------------------
+# Create a container to work with.
+#[user@desktop]
+
+    source "${HOME:?}/aglais.env"
+
+    agcolour=red
+    agproxymap=3000:3000
+
+    clientname=ansibler-${agcolour}
+    cloudname=iris-gaia-${agcolour}
+    configname=zeppelin-26.43-spark-6.26.43
+
+    podman run \
+        --rm \
+        --tty \
+        --interactive \
+        --name     "${clientname:?}" \
+        --hostname "${clientname:?}" \
+        --publish  "${agproxymap:?}" \
+        --env "cloudname=${cloudname:?}" \
+        --env "configname=${configname:?}" \
+        --env "SSH_AUTH_SOCK=/mnt/ssh_auth_sock" \
+        --volume "${SSH_AUTH_SOCK:?}:/mnt/ssh_auth_sock:rw,z" \
+        --volume "${HOME:?}/clouds.yaml:/etc/openstack/clouds.yaml:ro,z" \
+        --volume "${AGLAIS_CODE:?}/deployments:/deployments:ro,z" \
+        ghcr.io/wfau/atolmis/ansible-client:2022.07.25 \
+        bash
+
+
+# -----------------------------------------------------
+# Deploy everything.
+#[root@ansibler]
+
+    time \
+	source /deployments/hadoop-yarn/bin/deploy.sh
+
+
+	> real	77m19.087s
+	> user	10m18.075s
+	> sys	2m45.715s
+
+
+# -----------------------------------------------------
+# Import our live users.
+#[root@ansibler]
+
+    source /deployments/zeppelin/bin/create-user-tools.sh
+
+    import-test-users
+
+# -----------------------------------------------------
+# Start NGINX so that we can access using public IP.
+#[fedorat@zeppelin]
+
+    sudo service nginx start
+
```

Comment:
Does this mean that by default Nginx is configured to use HTTP on port 80?

Reply:
Nginx is configured to use port 80 yes

Reply:
Without a redirect to HTTPS on 443 ?

Reply:
If the intention is to only provide HTTPS on port 443, then why are we leaving port 80 open and why is Nginx listening on it ? Normal practice is to redirect port HTTP requests on port 80 to HTTPS on port 443, but we don't have that.

Reply:
This would require setting up certificates for the domains you listed below, so that we can access on port 443 when running a test deploy right?